### DISCUSSION TITLE
MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster

### DISCUSSION THREAD

**Discussion by ThakurSaad** [vercel/vercel#8601] [Help] [answered]
I am getting this error. Description in the image below. And after reloading once/twice(sometimes more), I am getting the data perfectly fine. The error says - 
`One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist`
But my **Atlas cluster's** network access is set to `0.0.0.0/0`. That means it includes all IP addresses. So, why is this happening? What am I doing wrong? 
![image](https://user-images.githubusercontent.com/96883294/191526072-893cc0b6-b803-4f89-9028-6ddee982e75f.png)

**Comment by amyegan** [comment, marked as answer]
Hi @ThakurSaad! You said that you get the data after reloading, so it sounds like an intermittent problem. I know the error says that a common reason is IP, but that doesn't mean it's the only possible cause. Sometimes a request takes too long and a timeout looks the same as a failed connection.

If it's a persistent problem, then you might have a problem with your database credentials. In that case, I would recommend checking your username and password, or whatever credentials are being used for db connection, in your environment variables and ensure they are set correctly for your production and preview environments.

The [MongoDB Developer Community](https://www.mongodb.com/community/forums/) has likely seen similar issues before and would have some good suggestions. It might be worth asking there as well, if you have not already.

**Reply by ThakurSaad** [reply]
Hello, @amyegan! Thank you for your help. Have a nice day.
